src/attributes.jl

```diff
@@ -0,0 +1,195 @@
+"""
+Main structure for holding attributes, for theming plots etc!
+"""
+struct Attributes
+    # name, for better error messages!
+    name::String
+    # We dont have one node per value anymore, but instead one node
+    # that gets triggered on any setindex!, or whenever an input attribute node changes
+    # This makes it easier to layer Observable{Attributes}()
+    on_change::Node{Pair{Symbol, Any}}
```

Comment:
With this, would a re-render be forced whenever you update `scene.color`?  I'm not sure how this would work, would appreciate some clarification here.

Reply:
That never triggered a rerender in opengl, and still wont ;) But I guess will trigger one in other frameworks... But its essentially not different from the Attribute type before, just an optimized implementation!